realm/realm-library/src/main/cpp/util.cpp

```diff
@@ -50,7 +50,7 @@ void ConvertException(JNIEnv* env, const char* file, int line)
     catch (JavaExceptionThrower& e) {
         e.throw_java_exception(env);
     }
-    catch (bad_alloc& e) {
+    catch (std::bad_alloc& e) {
```

Comment:
Why the need for `std::`? Line 36 is including the name space.

Reply:
Core made some changes to exception handling, which seems to require this, but I am also unsure if this is good enough to catch the exceptions now thrown from Core

Reply:
@finnschiermer or @simonask ? ^

Reply:
If you have both `using namespace std` and `using namespace realm::util`, `bad_alloc` is ambiguous since I put a `util::bad_alloc` in Core. It captures a backtrace from the throw-site and puts it in the `what()` description.

You can safely use `std::bad_alloc` and expect to catch both `std::bad_alloc` and `util::bad_alloc` (the latter inherits from the former). :-)